pkg/config/config.go

```diff
@@ -195,13 +192,28 @@ func evalFuncFile() function.Function {
 	})
 }
 
-// LoadComponentConfigBody returns nil if no component with the given
```

Comment:
The problems I have with `LoadComponentConfigBody`:

- "Load" doesn't mean anything to me. I can interpret it as "read from disk", "unmarshal" and probably a bunch of other options.
- It's long.
- "Config" is redundant since we are in the `config` package ("don't stutter"...).

>By the function Component I would expect to get an instance of Component interface, not HCL body.

I don't think there is anything which hints that you would get a `components.Component` here. We are in a package named `config`.
By the way, I've been contemplating whether `hcl` would be a better name for the package. On one hand it makes it easier to distinguish between "cluster config" (i.e. HCL files) and platform config (i.e. a struct containing platform-specific knobs).

How would this look to you if it were `hcl.Component()` and maybe also `hcl.Cluster()`?

Reply:
Regarding the config/HCL naming for the package, I think ideally we should have a `config` package, which allows expressing Lokomotive configuration using Go code and this one could have a `hcl` package, which would handle reading HCL files and converting them to Go structs. This way, if we want, we can later on also add different formats of the configuration.

>"Load" doesn't mean anything to me. I can interpret it as "read from disk", "unmarshal" and probably a bunch of other options.

Hm, in this particular case, we don't even load or read anything, so yeah, the name could be improved :)

I have a feeling, that getting rid of `LoadConfig` method from `Component` interface would save us time, as we would remove this method completely and we just return the `Component` to the user (or `map[string]Component`).